### ISSUE TITLE
Recreating Model Performance

### ISSUE THREAD

**Issue by Aldo-Aditiya** [mlmed/torchxrayvision#28]
We tried to recreate your pretrained model in: [Link](https://github.com/mlmed/torchxrayvision/releases/download/v1/nih-pc-chex-mimic_ch-google-openi-kaggle-densenet121-d121-tw-lr001-rot45-tr15-sc15-seed0-best.pt)

Somehow we couldn't achieve the same performance. We were using open dataset (NIH, Padchest, MIMIC, RSNA, NIH_Google, Chexpert) based on your paper ["On the limits of cross-domain
generalization in automated X-ray prediction"](https://arxiv.org/abs/2002.02497) as training/validation/test set and we retrieve all the datasets in original size. 

As such, we have some questions that we would like to ask you:

1. Regarding your augmentation settings in: [datasets.py](https://github.com/mlmed/torchxrayvision/blob/master/torchxrayvision/datasets.py): Are they enough (cropping to 224x224 pixels, center crop, etc) to recreate the same model as your pretrain models? Should the Dataset be in RGB format or grayscale?

2. Did you use the same preprocessing and augmentation for every dataset?

3. We tried your code in: [mlmed/covid-severity](https://github.com/mlmed/covid-severity) using the models we created. We tested it with your pretrained model: [Link](https://github.com/mlmed/torchxrayvision/releases/download/v1/nih-pc-chex-mimic_ch-google-openi-kaggle-densenet121-d121-tw-lr001-rot45-tr15-sc15-seed0-best.pt). Then, we compared it with our models (we trained the NIH, Padchest, MIMIC, RSNA, NIH_Google, Chexpert from scratch) with default hyper-parameter and default augmentation settings from your code. Somehow we could not achieve the same performance. Do you have any suggestions on why we couldn't achieve the same performance as you do? Or we were missing some steps here?

4. Did you use any form of weight initialization or pretrained models?

**Comment by ieee8023** [comment]
Sorry for the delay! The code that was used to train those models started here: https://github.com/ieee8023/xray-generalization and then it was modified to be the torchxrayvision library. Maybe there are some clues there that I overlooked.

One thing that changed is that the default views are now specified in the dataset loaders to be just PA. Before it was not consistent when the models were trained (some included AP). Maybe loading the datasets with views=["PA","AP"] would help?

> Did you use any form of weight initialization or pretrained models?

This is the model init code: https://github.com/mlmed/torchxrayvision/blob/master/torchxrayvision/models.py#L141

All the training code is here: https://github.com/mlmed/torchxrayvision/blob/master/scripts/train_model.py and https://github.com/mlmed/torchxrayvision/blob/master/scripts/train_utils.py

**Comment by matteoguarrera** [comment]
I was not able to found the performances of the different model on the test set. So I'm starting to benchmark some of them using https://github.com/mlmed/torchxrayvision/blob/bab8997af4c37159d43d2b8d194b95c0926cff5a/scripts/train_utils.py#L218

Do you have a complete benchmark of all the models? If not, here some preliminary results

| Dataset | Testset Size 	| Loss 	| Avg AUC 	|
|-	|-	|-	|-	|
| CHEX 	| 100% 	| 9.3634 	| 0.7923 	|
| NIH 	| 100% 	| 11.3296 	| 0.7953 	|
| RSNA 	| 100% 	| 1.5950 	| 0.8872 	|
| PC 	| 9% 	| 12.0923 	| 0.8712 	|

**Comment by ieee8023** [comment]
@matteoguarrera I did a benchmark here using these models (it is why they were built): https://arxiv.org/abs/2002.02497
I'll add the main figure showing the performance to the readme.
I think the main takeaway is that the performance should be tracked based on each task and not as the average.

**Comment by ieee8023** [comment]
A larger benchmark of the models in the library is here: https://github.com/mlmed/torchxrayvision/blob/master/BENCHMARKS.md

Closing this for now. Please reopen if not resolved.
